lib/spack/spack/schema/packages.py

```diff
@@ -85,6 +87,8 @@
                 "additionalProperties": False,
                 "properties": {
                     "require": requirements,
+                    "prefer": prefer_and_conflict,
+                    "conflict": prefer_and_conflict,
```

Comment:
at the very least the conflict should have `when` and `message`

Reply:
Done in https://github.com/spack/spack/pull/41832/commits/e24b8d3794181daf6a85049c3d13a063ee3d8d1b

Reply:
can you also add a test with a when:?

Reply:
Done in https://github.com/spack/spack/pull/41832/commits/f02d2e84c7cde89ac397fed0da5ef6dc2aa29e06